app/assets/javascripts/discourse/routes/preferences-second-factor.js.es6

```diff
@@ -13,5 +13,46 @@ export default RestrictedUserRoute.extend({
 
   setupController(controller, model) {
     controller.setProperties({ model, newUsername: model.get("username") });
+  },
+
+  actions: {
+    willTransition(transition) {
+      this._super();
+
+      const controller = this.controllerFor("preferences/second-factor");
+      const currentUser = controller.get('currentUser');
+      const siteSettings = controller.get('siteSettings');
+
+      if (transition.targetName === "preferences.second-factor") {
+        // User wants to set up second factor authentcation.
+        return true;
+      }
+
+      if (!currentUser) {
+        // No user logged in.
```

Comment:
These comments are actually not that useful since the code pretty much describes what the behavior is. I'll do away with the comments in this case.